kafka-connect-neo4j/src/main/kotlin/streams/kafka/connect/sink/converters/Neo4jValueConverter.kt

```diff
@@ -2,18 +2,44 @@ package streams.kafka.connect.sink.converters
 
 import org.neo4j.driver.Value
 import org.neo4j.driver.Values
+import org.slf4j.LoggerFactory
+import java.math.BigDecimal
+import java.math.BigInteger
 import java.time.LocalTime
 import java.time.ZoneId
-import java.util.Date
+import java.util.*
 import java.util.concurrent.TimeUnit
 
 
 class Neo4jValueConverter: MapValueConverter<Value>() {
 
     companion object {
-        @JvmStatic private val UTC = ZoneId.of("UTC")
+        @JvmStatic
+        private val UTC = ZoneId.of("UTC")
     }
 
+    override fun convert(data: Any?): MutableMap<String, Value?> {
+        val dataManipulatedForNeo4j = if (data is Map<*, *>) {
+            data.mapValues {
+                val value = it.value
+                if (value is BigInteger) {
+                    try {
+                        value.longValueExact()
+                    } catch (e: java.lang.ArithmeticException) {
+                        value.toString()
+                    }
+                } else {
+                    value
+                }
+            }
+        } else {
+            data
+        }
+
+        return super.convert(dataManipulatedForNeo4j)
+    }
```

Comment:
questo perché?

Reply:
Nel caso in cui venga passata nel converter una Map in cui un valore è un BigInteger maggiore di Long.MAX_VALUE, devo necessariamente fare qui la conversione in stringa, altrimenti poi in `AbstractConverter.java` che non è modificabile, il long verrebbe castato in `void convertMap(final T result, Map value)` con ((BigInteger) fieldValue).longValue(), e quindi verrebbe passato un valore errato (cioè 1), mentre mettendolo prima a stringa andrebbe in `AbstractConverter.java` su `fieldValue instanceof String` e quindi assumerebbe un valore corretto.
Questo si piò fare solo nel caso in cui venga passata una Mappa, perché se venisse passato uno Struct, dobbiamo assumere che il valore passato sia al massimo uguale a `Long.MAX_VALUE`, perche in `convertStruct(final T result, Struct struct)` di AbstractConverter comunque andrebbe a `case INT64:` e farebbe il cast a (Long), quindi non si può convertire a stringa precedentemente.
Nel caso invece di BigDecimal, posso coprire entrambi i casi (Struct e Map), modificando come fatto l'`override fun setDecimalField()`.